client/securedrop_client/sdk/__init__.py

```diff
@@ -213,10 +214,15 @@ def _streaming_download(
                 contents = fobj.read()
                 fobj.close()
 
-                # Check for an error response
+                # Check for an error response first at the request level...
                 if contents[0:1] == b"{":
                     logger.error(f"Retry {retry}, received JSON error response.")
                     return self._handle_json_response(contents)
+                # ...and then at the retry level:
+                partial = contents[bytes_written_prev:bytes_written]
```

Comment:
Should/can we skip this check if `bytes_written_prev == 0` (a no-retry request)?

Reply:
I wondered about this.  In terms of control flow:

1. In a retry request:
    1. `bytes_written_prev < len(contents)` and
    2. `bytes_written = len(contents)`, so
    3. `len(partial) < len(contents)`, so
    4. `partial[0:1] != contents[0:1]` and is worth checking,
    5. unless we've already returned on line 220.
2. In a non-retry request:
    1. `bytes_written_prev = 0` and
    2. `bytes_written = len(contents)`, so
    3. `len(partial) == len(contents)`, so
    4. `partial[0:1] == contents[0:1]` and is indeed redundant,
    5. unless we've already returned on line 220.

So the "should" here for me comes down to: Is this check too expensive to do twice for a successful request?  Probably not.  But it *is* confusing, so let me make it explicit.

Reply:
Done in c5d0b87c75bab4defc324f2a6b8cfaea5789be5c.  I've kept the conditionals parallel between `contents` and `partial`, but let me know if you'd prefer a nested one for `partial`.